onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -32,6 +32,29 @@ static inline void IncrementIndex(std::vector<int64_t>& index, const std::vector
   }
 }
 
+// Combines the two previous functions.
+template <typename T>
```

Comment:
Could this take sizeof(T) as a parameter instead of being templatized on T?

Reply:
I'll remove it for the final PR. The optimization can be applied in many places in the same file.